File: backend/app/schemas/audit.py
from typing import Optional
from uuid import UUID
from datetime import datetime

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: UUID
    user_id: Optional[UUID] = None
    action: str
    resource: Optional[str] = None
    resource_id: Optional[str] = None
    details: Optional[str] = None
    ip_address: Optional[str] = None
    timestamp: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/audit.ts
import apiClient from './client';

export interface AuditLog {
  id: string;
  user_id: string | null;
  action: string;
  resource: string | null;
  resource_id: string | null;
  details: string | null;
  ip_address: string | null;
  timestamp: string;
}

export const auditApi = {
  list: async (params?: {
    user_id?: string;
    action?: string;
    resource?: string;
    skip?: number;
    limit?: number;
  }): Promise<AuditLog[]> => {
    const response = await apiClient.get('/audit-logs/', { params });
    return response.data;
  },
};
